### ISSUE TITLE
Cannot use std::transform with Json::Value

### ISSUE THREAD

**Issue by daravi** [open-source-parsers/jsoncpp#1253]
**Describe the bug**
This code:
```
Json::Value result;
result.resize(value.size());
std::transform(value.begin(), value.end(), result.begin(), [](const auto& val){ return (bool)val; });
```
throws a runtime error:
```
    @     0x7f3ef4699c35 __GI_abort
    @     0x7f3ef4699b09 __assert_fail_base.cold.0
    @     0x7f3ef46a7de6 __GI___assert_fail
    @           0x563a1f Json::Value::releasePayload()
    @           0x5638ec Json::Value::~Value()
```

I can see that it is reaching the default case with value equal to 0xb0
```
void Value::releasePayload() {
  switch (type()) {
  case nullValue:
  case intValue:
  case uintValue:
  case realValue:
  case booleanValue:
    break;
  case stringValue:
    if (isAllocated())
      releasePrefixedStringValue(value_.string_);
    break;
  case arrayValue:
  case objectValue:
    delete value_.map_;
    break;
  default:
    JSON_ASSERT_UNREACHABLE;
  }
}
```

However using a for loop works:
```
for (int i = 0; i < value.size(); ++i)
{
	result[i] = (bool)value[i];
}
```

**Expected behavior**
For `std::transform` version to work as well.

Using version 1.9.2

**Comment by daravi** [comment]
Sorry I haven’t been able to test this further I will try to do that tonight.

**Comment by BillyDonahue** [comment]
I found the problem.

resize of a Null `Json::Value` is broken. I don't know the details yet but this repro is 3 lines long.

```c++
JSONTEST_FIXTURE_LOCAL(IteratorTest, transform_issue1253_minimal) {
    Json::Value result;
    int n = 10;
    result.resize(n);
    JSONTEST_ASSERT_EQUAL(n, result.size());  // passes
    JSONTEST_ASSERT_EQUAL(n, std::distance(result.begin(), result.end()));  // fails, actual=1
}
```

You get a Value that thinks its an array with 10 elements, but it only has one.
I think resize() is autovivifying the tail element and not the elements in between.

https://github.com/open-source-parsers/jsoncpp/compare/master...BillyDonahue:issue_1253_repro

**Comment by BillyDonahue** [comment]
This was fixed by PR #1265
